tests/e2e/spec/tonedear.e2e.spec.js

```diff
@@ -0,0 +1,196 @@
+/* eslint-disable no-undef */
+import { By, until } from 'selenium-webdriver';
+import assert from 'assert';
+import paths from '../paths.js';
+
+const BROWSERSTACK = !!process.env.BROWSERSTACK_LOCAL_IDENTIFIER;
+const port = BROWSERSTACK ? '8099' : '8080';
+const tonedearBaseFile = BROWSERSTACK ? '/tests/zims/tonedear/tonedear.com_en_2024-09.zim' : paths.tonedearBaseFile;
+
+/**
+ * Run the tests
+ * @param {WebDriver} driver Selenium WebDriver object
+ * @param {Array} modes Array of modes to run the tests in
+ * @param {boolean} keepDriver Whether to keep the driver open after the tests have run
+ * @returns {Promise<void>} A Promise for the completion of the tests
+ */
+function runTests (driver, modes, keepDriver) {
+    let browserName, browserVersion;
+    driver.getCapabilities().then(function (caps) {
+        browserName = caps.get('browserName');
+        browserVersion = caps.get('browserVersion');
+        console.log('\nRunning Tonedear tests on: ' + browserName + ' ' + browserVersion);
+    });
+
+    // Set the implicit wait to 3 seconds
+    driver.manage().setTimeouts({ implicit: 3000 });
+
+    // Run in both jquery and serviceworker modes by default
+    if (!modes) {
+        modes = ['jquery', 'serviceworker'];
+    }
+
+    modes.forEach(function (mode) {
+        const serviceWorkerAPI = true;
```

Comment:
You always set serviceWorkerAPI as true here, and it can't be changed (it's a const). Instead, we need to use a let here, and then change it if the serviceWorkerAPI isn't available, which is the case in some browsers.